### PR TITLE
Analyze/test analysis-options page code excerpts

### PR DESCRIPTION
**Pull request by chalin**
Resolves the second item of #1803.

Staged at https://dart-dev-staging-1.firebaseapp.com/guides/language/analysis-options
Original https://dart.dev/guides/language/analysis-options

Followup issue: #1918.
Possible followup issue: #1913.

---

I've checked external links: 

```console
$ blc -fgi http://localhost:4000/guides/language/analysis-options
Getting links from: http://localhost:4000/guides/language/analysis-options
├───OK─── https://dartpad.dev/
├───OK─── http://dart-lang.github.io/linter/lints/
├───OK─── http://www.google.com/intl/en/policies/privacy/
├───OK─── https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html
├───OK─── https://pub.dev/
├───OK─── https://flutter.dev/
├───OK─── http://creativecommons.org/licenses/by/3.0/
├───OK─── https://api.dart.dev/
├───OK─── https://flutter.dev/docs/testing/debugging#the-dart-analyzer
├───OK─── https://pub.dev/packages/analyzer
├───OK─── https://github.com/dart-lang/site-www/tree/master/src/_guides/language/analysis-options.md
├───OK─── https://pub.dev/packages/pedantic
├───OK─── https://github.com/dart-lang/site-www/issues/new?title=%27Customizing%20static%20analysis%27%20page%20issue&body=%0APage%20URL:%20https://dartlang-org-dev.firebaseapp.com/guides/language/analysis-options%0D%0A%0APage%20source:%20https://github.com/dart-lang/site-www/tree/master/src/_guides/language/analysis-options.md%0D%0A%0A%0D%0A%0AFound%20a%20typo?%20You%20can%20fix%20it%20yourself%20by%20going%20to%20the%20page%20source%20and%20clicking%20the%20pencil%20icon.%20Or%20finish%20creating%20this%20issue.%0D%0A%0A%0D%0A%0ADescription%20of%20issue:
├───OK─── https://medium.com/dartlang/announcing-dart-2-5-super-charged-development-328822024970
├───OK─── https://pub.dev/packages/glob
├───OK─── https://medium.com/dartlang
├─BROKEN─ https://pub.dev/documentation/analyzer/latest/analyzer/TodoCode/TODO-constant.html (HTTP_404)
├───OK─── https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/error/error.dart
├─BROKEN─ https://pub.dev/documentation/analyzer/latest/analyzer/AnalysisOptionsWarningCode/ANALYSIS_OPTION_DEPRECATED-constant.html (HTTP_404)
├─BROKEN─ https://pub.dev/documentation/analyzer/latest/analyzer/StaticTypeWarningCode/INVALID_ASSIGNMENT-constant.html (HTTP_404)
├───OK─── https://github.com/dart-lang/linter#linter-for-dart
├───OK─── https://github.com/dart-lang/site-www
```

The broken links are due to the fact that the [analyzer documentation generation failed for the latest release](https://pub.dev/packages/analyzer/versions):

> Version | Uploaded | Documentation | Archive
> -- | -- | -- | --
> 0.38.2 | Aug 28, 2019 | failed |

### CODE REVIEW

File: src/_guides/language/analysis-options.md [hunk 1]
```diff
@@ -64,38 +88,43 @@ at the root of the package, in the same directory as the pubspec file.
...
   Tracking issue: https://github.com/dart-lang/sdk/issues/28385
   {% endcomment %}
-</aside>
+{{site.alert.end}}
 
 Here's a sample analysis options file:
 
-{% prettify yaml %}
+<?code-excerpt "analysis_options.yaml" from="include" remove="implicit-dynamic" retain="/^$|\w+:|- camel/" remove="http:"?>
+```yaml
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: Interesting... the formatting for yaml got much less vibrant.

File: src/_guides/language/analysis-options.md [hunk 2]
```diff
@@ -64,38 +88,43 @@ at the root of the package, in the same directory as the pubspec file.
...
     - camel_case_types
+```
 
-analyzer:
-#   exclude:
-#     - path/to/excluded/files/**
-{% endprettify %}
+The sample illustrates the most common top-level entries:
+
+- Use <code>include: <em>url</em></code> to
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: Nice, much easier to read now.

File: src/_guides/language/analysis-options.md [hunk 3]
```diff
@@ -271,74 +310,66 @@ You can also [change the severity of rules][].
...
-To suppress a specific rule on a specific line of code,
-preceed that line with an `ignore` comment:
-
-{% prettify dart %}
-// ignore: <linter rule>
-{% endprettify %}
-
-Here's example of ignoring code that causes a runtime error,
+To suppress a specific rule on a specific line of code, preceed that line with
+an `ignore` comment.Here's example of ignoring code that causes a runtime error,
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: .Here's -> . Here's an

File: src/_guides/language/analysis-options.md [hunk 4]
```diff
@@ -12,12 +12,37 @@ With the help of the analyzer, you can find
...
+void increment() {
+  if (count < 10) ; /*1*/
+  count++;
+}
+```
+
+{:.console-output}
+<?code-excerpt "analysis/analyzer-results.txt" retain="empty_statements" replace="/ at (lib|test)\/\w+\.dart:\d+:\d+/ at \/*1*\//g"?>
+```nocode
+lint • Avoid empty statements at /*1*/ • empty_statements
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: This part seems weird to me... you'd never see comment text in there. Can we be more realistic?

File: src/_guides/language/analysis-options.md [hunk 5]
```diff
@@ -12,12 +12,37 @@ With the help of the analyzer, you can find
...
+<blockquote class="ml-3" markdown="1">
+<?code-excerpt "analysis/lib/lint.dart (close_sinks)"?>
+```dart
+var _controller = StreamController<String>(); /*2*/
+```
+
+{:.console-output}
+<?code-excerpt "analysis/analyzer-results.txt" retain="close_sinks" replace="/ at (lib|test)\/\w+\.dart:\d+:\d+/ at \/*2*\//g"?>
+```nocode
+lint • Close instances of `dart.core.Sink` at /*2*/ • close_sinks
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: Ditto here.

File: src/_guides/language/analysis-options.md (hunk 4 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: Ok, I'll see if I can think of something. I have an idea; I'll try it out tomorrow.

File: src/_guides/language/analysis-options.md [hunk 6]
```diff
@@ -271,74 +310,66 @@ You can also [change the severity of rules][].
...
 
-To suppress a specific rule on a specific line of code,
-preceed that line with an `ignore` comment:
-
-{% prettify dart %}
-// ignore: <linter rule>
-{% endprettify %}
-
-Here's example of ignoring code that causes a runtime error,
+To suppress a specific rule on a specific line of code, preceed that line with
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: GLOBAL: preceed -> precede

Comment (kwalrath) [reply src/_guides/language/analysis-options.md]: Or maybe just "put an `ignore` comment above the line of code."

File: src/_guides/language/analysis-options.md (hunk 1 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: Markdown code blocks are processed statically and rendered using the `highlight` package styles (as opposed to the prettifier JS package). I haven't yet tweaked `highlight` package styles in support of YAML. If you agree that this isn't gating, I'd suggest addressing it in a followup PR (though, if I have time before Filip or John review it, I ca add the style changes to this PR too).

File: src/_guides/language/analysis-options.md (hunk 3 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: Oops, thanks for catching that!

File: src/_guides/language/analysis-options.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/language/analysis-options.md]: Definitely not gating. I'd noticed the difference in styles for awhile, but it's starting to annoy me more and more. :)

File: src/_guides/language/analysis-options.md (hunk 4 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: I've added line numbers and dash-underlined in red just like, say, VS Code does. Let me know if that works for you.

Comment (chalin) [reply src/_guides/language/analysis-options.md]: Updates done and staging server refreshed. PTAL @kwalrath

File: src/_guides/language/analysis-options.md (hunk 1 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: Followup issue: #1918

**Comment by chalin** [comment]
@kwalrath - I justed added the IDE-like analyzer-error highlighting to the implicit-casts example too. PTAL. It is in a separate commit, so I can easily take it out if you don't like it.

---

Here is what the error highlighting looks like for the two opening examples:
> Screen Shot 2019-09-18 at 16 08 56

--- 

And here is what it looks like for the implicit-casts example:
> Screen Shot 2019-09-18 at 16 10 36

File: src/_guides/language/analysis-options.md [hunk 7]
```diff
@@ -173,10 +173,10 @@ You can use the flags together or separately; both default to `true`.
   includes an implicit downcast that would be caught by this flag:
 
 {:.fails-sa}
-<?code-excerpt "analysis/lib/assignment.dart (implicit-downcast)"?>
-{% prettify dart %}
+<?code-excerpt "analysis/lib/assignment.dart (implicit-downcast)" replace="/(s = )(o)/$1[!$2!]/g"?>
+{% prettify dart class="linenums:10 analyzer" %}
```
Comment (kwalrath) [inline src/_guides/language/analysis-options.md]: Do we really need linenums here? It's not referred to in the output. (Should it be?)

Comment (chalin) [reply src/_guides/language/analysis-options.md]: Ok, I've dropped the linenums.

Comment (chalin) [reply src/_guides/language/analysis-options.md]: Committed. Staging server updated.

**Comment by kwalrath** [comment]
Looks great! Feel free to squash and merge this. (I didn't because I wasn't sure if the order of these commits was important.)

File: src/_guides/language/analysis-options.md (hunk 1 above)
Comment (chalin) [reply src/_guides/language/analysis-options.md]: Addressed by #1924

**Pull request merged**